monai/transforms/utility/dictionary.py

```diff
@@ -415,9 +417,44 @@ def __call__(self, data: Mapping[Hashable, NdarrayOrTensor]) -> Dict[Hashable, N
                 if split_key in d:
                     raise RuntimeError(f"input data already contains key {split_key}.")
                 d[split_key] = r
+
+                if self.update_meta:
+                    orig_meta = d.get(PostFix.meta(key), None)
+                    if orig_meta is not None:
+                        split_meta_key = PostFix.meta(split_key)
+                        d[split_meta_key] = deepcopy(orig_meta)
+                        dim = self.splitter.dim
+                        if dim > 0:  # don't update affine if channel dim
+                            shift = np.zeros_like(d[split_meta_key]["affine"])  # type: ignore
+                            shift[dim - 1, -1] = i * d[split_meta_key]["pixdim"][dim]  # type: ignore
+                            d[split_meta_key]["affine"] += shift @ d[split_meta_key]["affine"]  # type: ignore
```

Comment:
I think the shift should happen in the image space (untested):
```suggestion
                            shift = np.eye(len(d[split_meta_key]["affine"]))
                            shift[dim - 1, -1] = i
                            d[split_meta_key]["affine"] = d[split_meta_key]["affine"] @ shift
```

with the current PR's implementation when the axis is flipped in the physical<>image space transform, the shift should be -shift